drivers/mspi/mspi_nrfe.c

```diff
@@ -0,0 +1,649 @@
+/*
+ * Copyright (c) 2024, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define DT_DRV_COMPAT nordic_nrfe_mspi_controller
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/mspi.h>
+#include <zephyr/ipc/ipc_service.h>
+#if CONFIG_PM_DEVICE
+#include <zephyr/pm/device.h>
+#endif
+#include <drivers/mspi/nrfe_mspi.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mspi_nrfe, CONFIG_MSPI_LOG_LEVEL);
+
+#define MSPI_NRFE_NODE	   DT_DRV_INST(0)
+#define MAX_TX_MSG_SIZE	   (DT_REG_SIZE(DT_NODELABEL(sram_tx)))
+#define MAX_RX_MSG_SIZE	   (DT_REG_SIZE(DT_NODELABEL(sram_rx)))
+#define CONFIG_TIMEOUT_MS  100
+#define EP_SEND_TIMEOUT_MS 10
+
+#if CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+PINCTRL_DT_INST_DEFINE(0);
+#endif
+
+static struct ipc_ept ep;
+static size_t ipc_received;
+static uint8_t *ipc_receive_buffer;
+
+#if defined(CONFIG_MULTITHREADING)
+static K_SEM_DEFINE(ipc_sem, 0, 1);
+static K_SEM_DEFINE(conf_sem, 0, 1);
+static K_SEM_DEFINE(xfer_sem, 0, 1);
+static K_SEM_DEFINE(recv_sem, 0, 1);
+#else
+static volatile uint32_t ipc_sem;
+static volatile uint32_t conf_sem;
+static volatile uint32_t xfer_sem;
+static volatile uint32_t recv_sem;
+#endif
+
+static struct gpio_dt_spec ce_gpios[] = MSPI_CE_GPIOS_DT_SPEC_GET(MSPI_NRFE_NODE);
+
+#define MSPI_CONFIG                                                                                \
+	{                                                                                          \
+		.channel_num = 0,                                                                  \
+		.op_mode = DT_PROP_OR(MSPI_NRFE_NODE, op_mode, MSPI_OP_MODE_CONTROLLER),           \
+		.duplex = DT_PROP_OR(MSPI_NRFE_NODE, duplex, MSPI_FULL_DUPLEX),                    \
+		.dqs_support = DT_PROP_OR(MSPI_NRFE_NODE, dqs_support, false),                     \
+		.ce_group = (struct gpio_dt_spec *)ce_gpios,                                       \
+		.num_ce_gpios = ARRAY_SIZE(ce_gpios),                                              \
+		.num_periph = DT_CHILD_NUM(MSPI_NRFE_NODE),                                        \
+		.max_freq = DT_PROP(MSPI_NRFE_NODE, clock_frequency),                              \
+		.re_init = true,                                                                   \
+		.sw_multi_periph = false,                                                          \
+	}
+
+struct mspi_nrfe_data {
+	struct mspi_xfer xfer;
+	struct mspi_dev_id dev_id;
+	struct mspi_dev_cfg dev_cfg;
+};
+
+static struct mspi_nrfe_data dev_data;
+
+struct mspi_nrfe_config {
+	struct mspi_cfg mspicfg;
+#if CONFIG_PINCTRL
```

Comment:
Is it valid to use this driver without pinctrl? My understanding that ownership of pin will not be configured then.